### ISSUE TITLE
Bug: [array-type] the config `simple-array` should be disallowed by the schema

### ISSUE THREAD

**Issue by Gotos** [typescript-eslint/typescript-eslint#6852]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.0.2&sourceType=module&code=MYewdgzgLgBAhgLhgCmgJwJZgOYwD4xgCuAtgEYCmaAlANoC6MAvDAwNwBQoksZSx5Kg2at6nbtBjAkAQTRo4ATwA86LLgIDKaAHwj2XcJIAms+UuVaqeluyA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6AQ2mncNqNPRgA2pERcA9tEgAaTJAAmiAGbsE+QZGSUAtsSS1O3QpAC+AXXBgTIE0A

### Repro Code

```typescript
const a: (string | number)[] = [];
const b: number[] = [];
const c: Array<string | number> = [];
const d: Array<number> = [];
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  "rules": {
    "@typescript-eslint/array-type": ["error", {"default": "simple-array"}]
  }
};
```

### Expected Result

The constants a and d should fail the 'array-simple' rule and thus cause the following errors:

a: `Array type using 'T[]' is forbidden for non-simple types. Use 'Array<T>'`
d: `Array type using 'Array<number>' is forbidden for simple types. Use 'number[]' instead.`

### Actual Result

All four constants fail the rule and cause errors. This means for example that neither `number[]` nor `Array<number>` is allowed, making it impossible to annotate array types without causing eslint error, if the "array-type" rule is configured with "simple-array".

**Comment by domdomegg** [comment]
I'm going to try to pick this one up. Also including in the scope of this the other validation of the array-type rule in general (#6892).

**Comment by bradzacher** [comment]
I had #6896 ready last night but I didn't push it because in fixing it I got side-tracked looking into fixing something tangentially related in `json-schema-to-typescript`.
Both PRs work together, @domdomegg - mine fixes the minimal schemas that need fixing and yours adds tests and tweaks other schemas, so no need to abandon or anything.
